kube-apiserver/tls-config.md

```diff
@@ -0,0 +1,366 @@
+	// to maximumVersion. If minimumVersion and maximumVersion are unset,
+	// the minimum version is determined by the TLS security profile type.
+	MinimumVersion TLSProtocolVersion `json:"minimumVersion"`
+	// maximumVersion enforces use of the specified TLSProtocolVersion or older
+	// that are negotiated during the TLS handshake. maximumVersion must be higher
+	// than or equal to minimumVersion.
+	//
+	// If unset and minimumVersion is set, maximumVersion will be set
+	// to minimumVersion. If minimumVersion and maximumVersion are unset,
+	// the maximum version is determined by the TLS security profile type.
+	MaximumVersion TLSProtocolVersion `json:"maximumVersion"`
+}
+
+// TLSProtocolVersion is a way to specify the protocol version used for TLS connections.
+// Protocol versions are based on the following most common TLS configurations:
+//
+//   https://ssl-config.mozilla.org/
+//
+// Note that SSLv3.0 is not a supported protocol version due to well known
+// vulnerabilities such as POODLE: https://en.wikipedia.org/wiki/POODLE
+type TLSProtocolVersion string
+
+const (
+	// TLSv1.0 is version 1.0 of the TLS security protocol.
+	VersionTLS10 TLSProtocolVersion = "TLSv1.0"
+	// TLSv1.1 is version 1.1 of the TLS security protocol.
+	VersionTLS11 TLSProtocolVersion = "TLSv1.1"
+	// TLSv1.2 is version 1.2 of the TLS security protocol.
+	VersionTLS12 TLSProtocolVersion = "TLSv1.2"
+	// TLSv1.3 is version 1.3 of the TLS security protocol.
+	VersionTLS13 TLSProtocolVersion = "TLSv1.3"
+)
+
+// DHParamSize sets the maximum size of the Diffie-Hellman parameters used for
+// generating the ephemeral/temporary Diffie-Hellman key.
+type DHParamSize string
+
+const (
+	// 1024 is a Diffie-Hellman parameter of 1024 bits.
+	DHParamSize1024 DHParamSize = "1024"
+	// 2048 is a Diffie-Hellman parameter of 2048 bits.
+	DHParamSize2048 DHParamSize = "2048"
+)
+
+// TLSProfiles Contains a map of TLSProfileType names to TLSProfileSpec.
+//
+// NOTE: The caller needs to make sure to check that these constants are valid for their binary. Not all
+// entries map to values for all binaries.  In the case of ties, the kube-apiserver wins.  Do not fail,
+// just be sure to whitelist only and everything will be ok.
+// TODO write the specifics of the ciphers, that's just too long to place in this KEP, it is in the openshift/api pull.
+var TLSProfiles = map[TLSProfileType]*TLSProfileSpec{
+	TLSProfileOldType: {
+		Ciphers: []string{
+		},
+		TLSVersion: TLSVersion{
+			MaximumVersion: VersionTLS10,
+			MinimumVersion: VersionTLS12,
+		},
+		DHParamSize: DHParamSize1024,
+	},
+	TLSProfileIntermediateType: {
+		Ciphers: []string{
+		},
+		TLSVersion: TLSVersion{
+			MaximumVersion: VersionTLS12,
+			MinimumVersion: VersionTLS12,
+		},
+		DHParamSize: DHParamSize2048,
+	},
+	TLSProfileModernType: {
+		Ciphers: []string{
+		},
+		TLSVersion: TLSVersion{
+			MaximumVersion: VersionTLS12,
+			MinimumVersion: VersionTLS12,
+		},
+		DHParamSize: DHParamSize2048,
```

Comment:
According to https://wiki.mozilla.org/Security/Server_Side_TLS#Modern_compatibility `DHParamSize` is nil.